### PR TITLE
Update NPC lanechange logic

### PR DESCRIPTION
**Pull request by hakuturu583**
Signed-off-by: Masaya Kataoka <ms.kataoka@gmail.com>

## Types of PR

- [ ] New Features
- [x] Upgrade of existing features
- [ ] Bugfix

## Description
NPC cannot lane change backwards.
## How to review this PR.
please check passing all CI tests.

### CODE REVIEW

File: simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+namespace traffic_simulator
+{
+namespace math
+{
+const geometry_msgs::msg::Pose getRelativePose(
+  const geometry_msgs::msg::Pose & from, const geometry_msgs::msg::Pose & to);
+}  // namespace math
+}  // namespace traffic_simulator
+
+#endif  // TRAFFIC_SIMULATOR__MATH__TRANSFORM_HPP_
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp:29]: PLEASE fix "No newline at end of this".

File: simulation/traffic_simulator/src/math/transform.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  geometry_msgs::msg::Pose ret;
+  {
+    tf2::toMsg(tf_delta, ret);
+  }
+
+  return ret;
+}
+}  // namespace math
+}  // namespace traffic_simulator
```
Comment (yamacir-kit) [inline simulation/traffic_simulator/src/math/transform.cpp:63]: PLEASE FIX "No newline at end of this".

**Review by yamacir-kit — CHANGES_REQUESTED** [review]
Please fix some warnings from GitHub.

File: simulation/traffic_simulator/src/math/transform.cpp (hunk 2 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/src/math/transform.cpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/372/commits/e408a9a5425d1c1e84f9a926594993abee52f75b

File: simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/traffic_simulator/include/traffic_simulator/math/transfrom.hpp]: fixed in https://github.com/tier4/scenario_simulator_v2/pull/372/commits/e408a9a5425d1c1e84f9a926594993abee52f75b

**Review by yamacir-kit — APPROVED** [review]

**Pull request merged**